tensorpipe/transport/shm/loop.cc

```diff
@@ -222,6 +231,40 @@ void Loop::join() {
   done_ = true;
   wakeup();
   thread_.join();
+  notificationMonitor_.join();
+}
+
+void Loop::notificationMonitor() {
+  while (!done_) {
+    uint32_t token;
+    auto ret = notification_->copy(token);
+    if (-ret == ENODATA) {
+      sched_yield();
```

Comment:
I was going to complain loudly about the busy waiting but this makes me kinda ok with it.

Reply:
We can still do better by adding a mutex and condition variable to the ring buffer header and having producers of these notifications signal there. Then this thread can be suspended if nothing is happening.